io-backend/src/main/java/agh/io/iobackend/service/UserService.java

```diff
@@ -32,4 +35,18 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
         return user.map(UserDetailsImpl::new).get();
     }
 
+    public User getCurrentUser(){
+        return userRepository.getById(getCurrentUserId());
+    }
+
+    public Long getCurrentUserId() {
```

Comment:
Czyli jeżeli okaże się że jesteśmy `AnonymousAuthenticationToken`, to getter zwróci null?

Reply:
tak, jeśli użytkownik nie jest zalogowany to chcemy zwrócić stąd nulla